pkg/edgeview/src/system.go

```diff
@@ -0,0 +1,1022 @@
+	if err != nil {
+		return
+	}
+	for _, s := range jfiles {
+		retbytes, _ := ioutil.ReadFile(s)
+		status := strings.TrimSuffix(string(retbytes), "\n")
+		appuuid := doAppNet(status, "", true)
+		retbytes, err := ioutil.ReadFile("/run/domainmgr/DomainMetric/"+appuuid+".json")
+		if err == nil {
+			var metric types.DomainMetric
+			_ = json.Unmarshal(retbytes, &metric)
+			fmt.Printf("    CPU: %d, Used Mem(MB): %d, Avail Mem(BM): %d\n",
+				metric.CPUTotalNs, metric.UsedMemory, metric.AvailableMemory)
+		}
+
+		retbytes, err = ioutil.ReadFile("/run/zedmanager/DomainConfig/"+appuuid+".json")
+		if err != nil {
+			continue
+		}
+		printColor("\n  - vnc/log info:", colorGREEN)
+		var config types.DomainConfig
+		_ = json.Unmarshal(retbytes, &config)
+		fmt.Printf("    VNC enabled: %v, VNC display id: %d, Applog disabled: %v\n",
+			config.EnableVnc, config.VncDisplay, config.DisableLogs)
+	}
+}
+
+func getDataStore() {
+	jfiles, err := listJSONFiles("/run/zedagent/DatastoreConfig")
+	if err != nil {
+		return
+	}
+
+	printColor(" - DataStore:", colorCYAN)
+	for _, l := range jfiles {
+		retbytes1, err := ioutil.ReadFile(l)
+		if err != nil {
+			continue
+		}
+		var data types.DatastoreConfig
+		_ = json.Unmarshal(retbytes1, &data)
+		if data.IsCipher {
+			fmt.Printf("   Cipher Context ID: %s, Cipher Hash: %s\n",
+				data.CipherContextID, base64.StdEncoding.EncodeToString(data.ClearTextHash))
+		}
+		fmt.Printf("\n   FQDN: %s, Path: %s, DS Type: %s, Is Cipher: %v\n", data.Fqdn, data.Dpath, data.DsType, data.IsCipher)
+		if len(data.DsCertPEM) > 0 {
+			for _, c := range data.DsCertPEM {
+				printCert(c)
+			}
+		}
+	}
+}
+
+func getModel() {
+	printTitle("Model:", colorCYAN, false)
+	_, _ = runCmd("spec.sh", true)
+}
+
+func getHW() {
+	printTitle("HW:", colorCYAN, false)
+	err := addPackage("/usr/sbin/lshw", "lshw")
+	if err != nil {
+		fmt.Printf("add package: %v\n", err)
+		return
+	}
+	_, _ = runCmd("lshw -json", true)
+}
+
+func getLastReboot() {
+	files, err := ioutil.ReadDir("/persist/log")
+	if err != nil {
+		fmt.Printf("failed to get to /persist/log\n")
+		return
+	}
+
+	for _, l := range files {
+		var rebootFile string
+		if strings.Contains(l.Name(), "reboot-reason.log") {
+			rebootFile = "reboot-reason.log"
+		} else if strings.Contains(l.Name(), "reboot-stack.log") {
+			rebootFile = "reboot-stack.log"
+		} else {
+			continue
+		}
+		printTitle(rebootFile, colorBLUE, false)
+		if strings.Contains(rebootFile, "reason") {
+			readAFile("/persist/log/"+rebootFile, -5)
+		} else {
+			readAFile("/persist/log/"+rebootFile, 0)
+		}
+	}
+
+	files, err = ioutil.ReadDir("/persist/newlog/panicStacks")
+	if err != nil {
+		fmt.Printf("failed to get to /persist/log\n")
+		return
+	}
+
+	for _, l := range files {
+		if strings.Contains(l.Name(), "pillar-panic-stack.") {
+			fields := strings.Fields(l.Name())
+			n := len(fields)
+			retbytes, err := ioutil.ReadFile("/persist/newlog/panicStacks/"+fields[n-1])
```

Comment:
Does this need a check for n >=1 to avoid referencing fields[-1]?

Reply:
added a check.